### PR TITLE
Fix SSH tunnel credentials not being validated

### LINKED ISSUE #5016: SSH tunnel credentials not being validated
It's possible to add a database with invalid SSH tunnel credentials. The database doesn't sync and it isn't usable.

### PR DESCRIPTION
**Pull request by thearthur**
Fix a bug in validating ssh tunnel parameters in generic-sql, druid, and oracle drivers.

Resolves #5016  and adds tests to all the drivers with ssh tunnels that
implement their own can-connect?

### CODE REVIEW

File: test/metabase/driver/druid_test.clj [hunk 1]
```diff
@@ -5,11 +5,13 @@
...
              [timeseries-query-processor-test :as timeseries-qp-test]
              [util :as u]]
+            [expectations :refer [expect]]
             [metabase.models.metric :refer [Metric]]
             [metabase.query-processor.expand :as ql]
             [metabase.test.data :as data]
             [metabase.test.data.datasets :as datasets :refer [expect-with-engine]]
-            [toucan.util.test :as tt]))
+            [toucan.util.test :as tt]
+            [metabase.driver :as driver]))
```
Comment (camsaul) [inline test/metabase/driver/druid_test.clj]: https://github.com/metabase/metabase/wiki/Metabase-Clojure-Style-Guide#keep-requires-ordered

File: test/metabase/driver/mongo/util_test.clj [hunk 2]
```diff
@@ -1,7 +1,8 @@
 (ns metabase.driver.mongo.util-test
   (:require [expectations :refer :all]
             metabase.driver.mongo.util
-            [metabase.test.util :as tu])
+            [metabase.test.util :as tu]
+            [metabase.driver :as driver])
```
Comment (camsaul) [inline test/metabase/driver/mongo/util_test.clj]: https://github.com/metabase/metabase/wiki/Metabase-Clojure-Style-Guide#keep-requires-ordered

I thought you had Emacs set up to automatically call `cljr-clean-ns`?

**Review by camsaul — APPROVED** [review]
Ok, this looks good besides the `:require` statements being out-of-order

File: test/metabase/driver/mongo/util_test.clj (hunk 2 above)
Comment (thearthur) [inline test/metabase/driver/mongo/util_test.clj]: I thought so too, so i have been depending on it ... will fix.

**Pull request merged**